### DISCUSSION TITLE
Default Theme - better initial values + access to `ChartStyle.fromDefaultColors`

### DISCUSSION THREAD

**Discussion by MFlisar** [patrykandpatrick/vico#456] [Ideas]
I offer my users to use custom themes inside my app - this means using `isSystemInDarkTheme()` is not always correct in my case and this leads to following 2 suggestions:

**Suggestion 1**

I think it would be nice to adjust following function:

https://github.com/patrykandpatrick/vico/blob/45f1b08497521ee50603251e8e6d70564fbf1a28/vico/compose/src/main/java/com/patrykandpatrick/vico/compose/style/ChartStyle.kt#L253

I would change the hard coded colors inside `DefaultColors.Dark` and `DefaultColors.Light` to something that is derived from `MaterialTheme.colorScheme` and/or `LocalContentColor`. What do you think about that?

**Suggestion 2**

In my use case where the `isSystemInDarkTheme()` does not fit the real applied theme, I would like to have access to the `ChartStyle.fromDefaultColors` function where I can pass in my own `isDark` flag.

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestions.

1. The `m2ChartStyle` and `m3ChartStyle` functions use colors from the app’s Material Design theme. Please consider using one of these. Note that they have their own modules, as described in the wiki. The `compose` module doesn’t depend on any Material Design libraries.
2. Since `DefaultColors` is public, it would make sense for `ChartStyle.fromDefaultColors` to be public too. We’ll look into it. In the meantime, you can copy over the function’s definition, which is relatively simple, to your project.

**Reply by MFlisar** [reply]
`m3ChartStyle` is all I need, I must have overseen that, this solves all my problems.

Regarding the `DefaultColors`, its up to you if you expose that, but as far as I understand, those hard coded default colors are just there to make the compose core independent of m2 and m3 so in the end, I assume, its fine like it is currently...

**Reply by patrickmichalik** [reply]
I’m glad to hear `m3ChartStyle` did the trick. Yup, `m2ChartStyle` and `m3ChartStyle` are in their own modules so that `compose` is independent of Material Design.

**Comment by Gowsky** [comment]
This has been addressed in Vico 2. We've chosen a different approach and `DefaultColors` are no longer public.
